src/commands/app/delete/action.js

```diff
@@ -72,46 +76,53 @@ class DeleteActionCommand extends BaseCommand {
     ])
 
     if (!flags.yes && !resConfirm.deleteAction) {
-      this.log('aborting..')
+      this.error('aborting..')
     }
-    actionsToBeDeleted.forEach(a => {
+
+    actionsToBeDeleted.forEach(action => {
       // remove action files
-      const folder = fs.statSync(a.path).isFile() ? path.dirname(a.path) : a.path
+      const folder = fs.statSync(action.path).isFile() ? path.dirname(action.path) : action.path
       fs.removeSync(folder)
       aioLogger.debug(`deleted '${folder}'`)
       // delete test files
       // NOTE: those paths are not always correct, but removeSync doesn't throw in case the file does not exist
-      const pathToE2eTests = path.join('e2e', a.actionsDir, a.actionName + '.e2e.js')
-      const pathToUnitTests = path.join('test', a.actionsDir, a.actionName + '.test.js')
-      fs.removeSync(pathToE2eTests)
-      aioLogger.debug(`deleted '${pathToE2eTests}'`)
-      fs.removeSync(pathToUnitTests)
-      aioLogger.debug(`deleted '${pathToUnitTests}'`)
+      try {
+        // coverage missing start
+        const pathToE2eTests = path.join('e2e', action.actionsDir, action.actionName + '.e2e.js')
+        const pathToUnitTests = path.join('test', action.actionsDir, action.actionName + '.test.js')
+        fs.removeSync(pathToE2eTests)
+        aioLogger.debug(`deleted '${pathToE2eTests}'`)
+        fs.removeSync(pathToUnitTests)
+        aioLogger.debug(`deleted '${pathToUnitTests}'`)
 
-      // delete manifest action config
-      deleteUserConfig(a.configData)
+        // delete manifest action config
+        deleteUserConfig(action.configData)
+        // coverage missing end
+      } catch (e) {
+        console.log('error', e)
+      }
 
-      this.log(chalk.green(`✔ Deleted '${a.name}'`))
+      this.log(chalk.green(`✔ Deleted '${action.name}'`))
     })
     this.log(chalk.bold(chalk.green(
-          `✔ Successfully deleted action(s) '${actionsToBeDeleted.map(a => a.name)}'` + EOL +
-          '  => please make sure to cleanup associated dependencies and to undeploy any deleted actions'
+      `✔ Successfully deleted action(s) '${actionsToBeDeleted.map(a => a.name)}'` + EOL +
+      '  => please make sure to cleanup associated dependencies and to undeploy any deleted actions'
     )))
   }
 
   getAllActions (config) {
     const actions = []
     const actionsByImpl = {}
     Object.entries(config.all).forEach(([implName, implConfig]) => {
-      if (implConfig.app.hasBackend) {
+      if (implConfig.app && implConfig.app.hasBackend) {
```

Comment:
nitpick: implConfig.app is always defined, so no need to have an extra branch to test ?

Reply:
it was not in the tests I wrote ...

Reply:
one day when everyone is on node 14 we can use optional?.chaining